File: ui/social-network/src/services/interaction/voteService.ts
import api from "../../axios/interceptor"
const path = "/interactions"
export interface VoteUpdateReq {
    objectId: number,
    objectType: "POST" | "COMMENT",
    voteType: "DOWN" | "UP"
}
class VoteService {
    updateVote(req: any) {
        return api.put(`${path}/votes`, req)
    }

    checkVote(objectId: any, objectType: "POST" | "COMMENT") {
        return api.get(`${path}/votes/check/${objectType}/${objectId}`)

    }
    count(objId: any, objType: "POST" | "COMMENT") {
        return api.get(`${path}/votes/scores/${objType}/${objId}`)
    }
}
export default new VoteService()

File: backend/services/interaction-service/interaction-biz/src/main/java/viosmash/services/VoteService.java
package viosmash.services;

import viosmash.controller.vote.vo.VoteUpdateReqVO;
import viosmash.interaction.enums.ObjectType;

public interface VoteService {
    void updateVote(Long userId, VoteUpdateReqVO req);
    int checkVote(Long userId, Long objectId, ObjectType objectType);
    int count(Long objId, ObjectType objType);
}
